pkg/user/appointee/can_call.go

```diff
@@ -0,0 +1,153 @@
+package appointee
+
+import (
+	"context"
+	"github.com/Layr-Labs/eigenlayer-cli/pkg/internal/common"
+	"github.com/Layr-Labs/eigenlayer-cli/pkg/internal/common/flags"
+	"github.com/Layr-Labs/eigenlayer-cli/pkg/telemetry"
+	"github.com/Layr-Labs/eigenlayer-cli/pkg/utils"
+	"github.com/Layr-Labs/eigensdk-go/chainio/clients/elcontracts"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	eigenSdkUtils "github.com/Layr-Labs/eigensdk-go/utils"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/urfave/cli/v2"
+)
+
+type UserCanCallReader interface {
+	UserCanCall(
+		ctx context.Context,
+		userAddress gethcommon.Address,
+		callerAddress gethcommon.Address,
+		target gethcommon.Address,
+		selector [4]byte,
+	) (bool, error)
+}
+
+func CanCallCmd() *cli.Command {
+	canCallCmd := &cli.Command{
+		Name:      "can-call",
+		Usage:     "user appointee can-call <AccountsAddress> <CallerAddress> <TargetAddress> <Selector>",
+		UsageText: "Checks if a user has a specific permission.",
+		Description: `
+		Checks if a user has a specific permission.
+		`,
+		Action: func(c *cli.Context) error {
+			return canCall(c)
+		},
+		After: telemetry.AfterRunAction(),
+		Flags: []cli.Flag{
+			&AccountAddressFlag,
+			&CallerAddressFlag,
+			&TargetAddressFlag,
+			&SelectorFlag,
+			&PermissionControllerAddressFlag,
+			&flags.NetworkFlag,
+			&flags.EnvironmentFlag,
+			&flags.ETHRpcUrlFlag,
+		},
+	}
+
+	return canCallCmd
+}
+
+func canCall(cliCtx *cli.Context) error {
+	ctx := cliCtx.Context
+	logger := common.GetLogger(cliCtx)
+
+	config, err := readAndValidateUserConfig(cliCtx, logger)
+	if err != nil {
+		return eigenSdkUtils.WrapError("failed to read and validate user can call config", err)
+	}
+
+	elReader, err := getEigenLayerReader(cliCtx, logger, config)
+	if err != nil {
+		return err
+	}
+
+	var result bool
+	result, err = elReader.UserCanCall(ctx, config.UserAddress, config.CallerAddress, config.Target, config.Selector)
```

Comment:
```suggestion
	result, err := elReader.UserCanCall(ctx, config.UserAddress, config.CallerAddress, config.Target, config.Selector)
```

Reply:
`err` is defined above, so you can't use `:=`.

Reply:
doesn't matter if `result` is a new variable. you can always hav `:=` if at least on one of the variable is undefined on the left side.